### ISSUE TITLE
MeshConfig network gateway address does not accept DNS address

### ISSUE THREAD

**Issue by anmolagrawal23** [istio/istio#32935]
When provided DNS address in meshconfig's network gateway address it throws an error.

Istio Version: 1.9.5

MeshConfig:
```
  meshNetworks: |-
    networks:
      network2:
        endpoints:
        - fromRegistry: cluster2
        gateways:
        - address: istio-eastwest-gateway.example.com
          port: 32121
```

Error:
```
2021-05-19T04:33:13.884831Z     info    failed to read mesh networks configuration from "/etc/istio/config/meshNetworks": 1 error occurred:
        * invalid network network2: istio-eastwest-gateway.example.com is not a valid IP
```

As per the [IstioNetworkGateway documentation](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#Network-IstioNetworkGateway), either an IP address or a DNS name can be provided in the configuration but it throws error for DNS name.
  
Steps to reproduce:
In mesh configuration, provide DNS address for Network gateway configuration.

[X] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Comment by stevenctl** [comment]
We should update the documentation until #29359 is resolved. 

Also, MeshNetworks is deprecated and we try to rely on the`topology.istio.io/network` label on the k8s Service for your k8s gateway. `istiod` will see this label and configure the gateway for cross-network automatically. I'm assuming based on the port number you're doing something special with NodePort, which _should_ be supported by our auto-detection.

**Comment by stevenctl** [comment]
related: https://github.com/istio/istio/issues/30345#event-4723343712

**Comment by anmolagrawal23** [comment]
We are trying multi-primary setup.
Our clusters are private and does not support LoadBalancer type services. We have created NodePort type service and use  NodeIP or external CName for the ingress to the cluster.
We got it working by adding Node IP address in the mesh network as per below configuration but instead we want to use CName (DNS address) so that it load balances to all the nodes.
```
  meshNetworks: |-
    networks:
      network2:
        endpoints:
        - fromRegistry: cluster2
        gateways:
        - address: 1.2.3.4
          port: 32121
```
If we add `topology.istio.io/network` label to the Gateway service, then will it work for this scenario?
How can we achieve this without using mesh networks.

**Comment by stevenctl** [comment]
A DNS based way wouldn't work, but Istio would include the addresses of each of the applicable nodes as cross-network gateways in EDS with that label.

**Comment by stevenctl** [comment]
Going to leave this closed, in favor of #29359
